NLSY/nlsy_lib.R

```diff
@@ -241,6 +241,121 @@ nlsy_get_highest_grade_completed_df = function()
     return(data)
 }
 
+#' Returns marriage status the last month of a given year
+nlsy_get_marstat_df = function()
+{
+    data = nlsydf |>
+        select(
+            id =        "PUBID_1997",
+            starts_with('MAR_STATUS_')
+        ) |>
+        pivot_longer(
+            starts_with('MAR_STATUS_'),
+            names_to=c('year', 'month'),
+            names_pattern='MAR_STATUS_(\\d\\d\\d\\d)\\.(\\d\\d)_XRND',
+            values_to='mar_status'
+        ) |>
+        mutate(
+            year=as.integer(year),
+            month=as.integer(month),
+            single=as.integer(mar_status == 'Never Married, Not Cohabitating'),
+            married=as.integer(mar_status == 'Married'),
+            divorced=as.integer(mar_status == 'Divorced'),
+            widowed=as.integer(mar_status == 'Widowed')
+        )
+
+    data = data |>
+        filter(month == 12) |>
+        group_by(id, year) |>
+        summarise(
```

Comment:
I wonder why we need `summarise` here, and `all` and `any` inside the call. Leaving only December in the sample and grouping by id and year creates one observation per group, so `all` and `any` can be dropped and `summarise` can be replaced by `mutate`.

That being said, perhaps there is a use for `all` and `any` in a little different context. We could check whether the December value is missing. If it is missing, we could then check the values of other months in the same year are missing. If there is a non-missing value in that year, we can use the one closest to December.

Reply:
Good point! Addressed this in the `fill()` edits for the comment below.